packages/devtools_app/lib/src/debugger/flutter/controls.dart

```diff
@@ -0,0 +1,209 @@
+                    ),
+                    _LeftBorder(
+                      child: DebuggerButton(
+                        title: 'Resume',
+                        icon: Icons.play_arrow,
+                        onPressed: isPaused ? controller.resume : null,
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+              const SizedBox(width: denseSpacing),
+              RoundedOutlinedBorder(
+                child: Row(
+                  children: [
+                    DebuggerButton(
+                      title: 'Step In',
+                      icon: Icons.keyboard_arrow_down,
+                      onPressed: canStep ? controller.stepIn : null,
+                    ),
+                    _LeftBorder(
+                      child: DebuggerButton(
+                        title: 'Step Over',
+                        icon: Icons.keyboard_arrow_right,
+                        onPressed: canStep ? controller.stepOver : null,
+                      ),
+                    ),
+                    _LeftBorder(
+                      child: DebuggerButton(
+                        title: 'Step Out',
+                        icon: Icons.keyboard_arrow_up,
+                        onPressed: canStep ? controller.stepOut : null,
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+              const Expanded(child: SizedBox(width: denseSpacing)),
+              RoundedOutlinedBorder(
+                child: Center(
+                  child: Padding(
+                    padding: const EdgeInsets.only(
+                        left: denseSpacing, right: borderPadding),
+                    child: BreakOnExceptionsControl(controller: controller),
+                  ),
+                ),
+              ),
+            ],
+          ),
+        );
+      },
+    );
+  }
+}
+
+class BreakOnExceptionsControl extends StatelessWidget {
+  const BreakOnExceptionsControl({
+    Key key,
+    @required this.controller,
+  }) : super(key: key);
+
+  final DebuggerController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder(
+      valueListenable: controller.exceptionPauseMode,
+      builder: (BuildContext context, String modeId, _) {
+        return DropdownButtonHideUnderline(
+          child: DropdownButton<ExceptionMode>(
+            value: ExceptionMode.from(modeId),
+            onChanged: (ExceptionMode mode) {
+              controller.setExceptionPauseMode(mode.id);
+            },
+            isDense: true,
+            items: ExceptionMode.modes.map((mode) {
+              return DropdownMenuItem<ExceptionMode>(
+                value: mode,
+                child: Text(mode.description),
+              );
+            }).toList(),
+            selectedItemBuilder: (BuildContext context) {
+              return ExceptionMode.modes.map((mode) {
```

Comment:
nit: consider using List comprehensions for this case instead of map followed by toList().
@pq this might be a reasonable general lint to encourage users to take advantage of list comprehensions if we think the list comprehension way is easier to read.
```dart
ExceptionMode.modes.map((mode) {
  return DropdownMenuItem<ExceptionMode>(
    value: mode,
    child: Text(mode.name),
  );
}).toList();
```
vs
```dart
[ for (mode in ExceptionMode.modes)
  DropdownMenuItem<ExceptionMode>(
    value: mode,
    child: Text(mode.name),
  )
]
```

Reply:
Neat.  https://github.com/dart-lang/linter/issues/2074 to track.

Reply:
done